lint/action.yaml

```diff
@@ -21,31 +19,15 @@ runs:
       with:
         github-personal-access-token: ${{ inputs.github-token }}
     - name: Checkout
-      uses: actions/checkout@v3
+      uses: actions/checkout@v4
       if: inputs.checkout-repo == 'true'
       with:
         fetch-depth: 0
-    - name: Setup tools
-      uses: open-turo/action-setup-tools@v1
-    # TODO: Configure .pypi credentials using pypi-token
-    - name: Install poetry
-      if: hashFiles('poetry.lock') != ''
-      shell: bash
-      run: ${GITHUB_ACTION_PATH}/../lib/poetry-install.sh
-    - name: Make sure poetry uses the current python version
-      if: hashFiles('poetry.lock') != ''
+    - name: Setup python
+      uses: actions/setup-python@v5
+    - name: Install pre-commit
+      run: pip install pre-commit
       shell: bash
-      run: |
-        # Use the version that pyenv gives as python version pyenv version-name
-        if ! command -v pyenv &>/dev/null; then
-          echo "pyenv not found, skipping"
-          exit 0
-        fi
-        PYTHON_VERSION_SET=$(pyenv version-name)
-        poetry env use $PYTHON_VERSION_SET
-    - name: Install pip dependencies
-      if: hashFiles('poetry.lock') == ''
+    - name: Run pre-commit
+      run: pre-commit run --show-diff-on-failure --color=always --all-files
```

Comment:
I still think we should be using here `open-turo/action-pre-commit` instead here by doing this change, you lost some functionality (like commit message linting)